### PR TITLE
CATROID-833 Missing tests for Catblocks integration

### PR DESCRIPTION
**Pull request by bprattes**
Add some basic tests for Catblocks in Catroid.
https://jira.catrob.at/browse/CATROID-833

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		onView(withText(R.string.catblocks_reorder)).check(doesNotExist());
+
+		onView(withText(R.string.backpack)).check(matches(isDisplayed()));
+		onView(withText(R.string.copy)).check(matches(isDisplayed()));
+		onView(withText(R.string.comment_in_out)).check(matches(isDisplayed()));
+		onView(withText(R.string.catblocks)).check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void testCorrect2DObjectDispayed() {
```
Comment (juliajulie95) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java:117]: Typo "Dispayed"

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	@Test
+	public void testReorderScript() {
+		openContextualActionModeOverflowMenu();
+		onView(withText(R.string.catblocks)).perform(click());
+
+		int loadCounter = 0;
+		// since rendering may take some time: try multiple times if finished
+		while (true) {
```
Comment (juliajulie95) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java:153]: While loop is duplicate of test above. Do we need to check this again? Maybe delete in this test, else delete test above.

**Review by juliajulie95 — CHANGES_REQUESTED** [review]

**Review by juliajulie95 — APPROVED** [review]

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+			} catch (Throwable e) {
+				// max. 2.5 seconds - then timeout
+				if (loadCounter >= 25) {
+					throw e;
+				}
+				try {
+					Thread.sleep(100);
+				} catch (InterruptedException ex) {
+					// ignore
+				}
```
Comment (wslany) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java:142]: This results in a lint warning => 
- https://jenkins.catrob.at/job/Catroid/job/PR-3888/14/analysis/ 
- https://jenkins.catrob.at/job/Catroid/job/PR-3888/14/analysis/source.01d88343-443b-496d-90eb-a6ac09189969/#140

**Review by wslany — CHANGES_REQUESTED** [review]
Please fix the warning.

**Review by juliajulie95 — APPROVED** [review]

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**